### ISSUE TITLE
Error: Cannot find module '/K:\GitXXXXXC\RSSHub\lib\routes/0818tuan/router.js'

### ISSUE THREAD

**Issue by HNGHTNLP** [DIYgod/RSSHub#14625]
### 路由地址

```routes
NOROUTE
```

### 完整路由地址

```fullroutes
NOROUTE
```

### 相关文档

https://docs.rsshub.app/zh/install#%E5%90%AF%E5%8A%A8

### 预期是什么？

正常启动RSSHub

### 实际发生了什么？

无法正常启动RSSHub
Linux正常
不清楚什么原因

### 部署

自建

### 部署相关信息

OS:Windows, Node:v20.11.1

### 额外信息

```shell
yarn run v1.22.21
$ tsx lib/index.ts
node:internal/modules/cjs/loader:1144
  const err = new Error(message);
              ^

Error: Cannot find module '/K:\GitXXXXXC\RSSHub\lib\routes/0818tuan/router.js'
Require stack:
- K:\GitXXXXXC\RSSHub\node_modules\directory-import\dist\src\import-modules.js
- K:\GitXXXXXC\RSSHub\node_modules\directory-import\dist\src\index.js
- K:\GitXXXXXC\RSSHub\lib\routes.ts
- K:\GitXXXXXC\RSSHub\lib\app.ts
- K:\GitXXXXXC\RSSHub\lib\index.ts
    at Module._resolveFilename (node:internal/modules/cjs/loader:1144:15)
    at a._resolveFilename (K:\GitXXXXXC\RSSHub\node_modules\tsx\dist\cjs\index.cjs:1:1729)
    at Module._load (node:internal/modules/cjs/loader:985:27)
    at Module.require (node:internal/modules/cjs/loader:1235:19)
    at require (node:internal/modules/helpers:176:18)
    at importModule (K:\GitXXXXXC\RSSHub\node_modules\directory-import\dist\src\import-modules.js:68:28)
    at Object.syncHandler [as sync] (K:\GitXXXXXC\RSSHub\node_modules\directory-import\dist\src\import-modules.js:21:34)    at importModules (K:\GitXXXXXC\RSSHub\node_modules\directory-import\dist\src\import-modules.js:84:40)
    at directoryImport (K:\GitXXXXXC\RSSHub\node_modules\directory-import\dist\src\index.js:17:45)
    at path (k:\GitXXXXXC\RSSHub\lib\routes.ts:13:17)
    at Object.<anonymous> (k:\GitXXXXXC\RSSHub\lib\routes.ts:31:8)
    at Module._compile (node:internal/modules/cjs/loader:1376:14)
    at Object.S (K:\GitXXXXXC\RSSHub\node_modules\tsx\dist\cjs\index.cjs:1:1292)
    at Module.load (node:internal/modules/cjs/loader:1207:32)
    at Module._load (node:internal/modules/cjs/loader:1023:12)
    at Module.require (node:internal/modules/cjs/loader:1235:19) {
  code: 'MODULE_NOT_FOUND',
  requireStack: [
    'K:\\GitXXXXXC\\RSSHub\\node_modules\\directory-import\\dist\\src\\import-modules.js',
    'K:\\GitXXXXXC\\RSSHub\\node_modules\\directory-import\\dist\\src\\index.js',
    'K:\\GitXXXXXC\\RSSHub\\lib\\routes.ts',
    'K:\\GitXXXXXC\\RSSHub\\lib\\app.ts',
    'K:\\GitXXXXXC\\RSSHub\\lib\\index.ts'
  ],
  includeSubdirectories: true,
  importMode: 'sync',
  importPattern: /router\.js$/,
  limit: Infinity,
  callerFilePath: '/',
  callerDirectoryPath: '',
  targetDirectoryPath: 'K:\\GitXXXXXC\\RSSHub\\lib\\routes',
  callback: undefined
}

Node.js v20.11.1
error Command failed with exit code 1.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
```

### 这不是重复的 issue

- [X] 我已经搜索了 [现有 issue](https://github.com/DIYgod/RSSHub/issues)，以确保该错误尚未被报告。

**Comment by HNGHTNLP** [comment]
set PUPPETEER_SKIP_CHROMIUM_DOWNLOAD=true
有关系吗

**Comment by huanfe1** [comment]
遇到了同样问题

**Comment by hamadawon** [comment]
same here

**Comment by lw99lw** [comment]
yarn run v1.22.21
$ cross-env NODE_ENV=production tsx lib/index.ts
node:internal/modules/cjs/loader:1144
  const err = new Error(message);
              ^

Error: Cannot find module '../assets/build/routes.json'
Require stack:
- D:\RSSHub\lib\registry.ts
- D:\RSSHub\lib\app.ts
- D:\RSSHub\lib\index.ts
    at Module._resolveFilename (node:internal/modules/cjs/loader:1144:15)
    at a._resolveFilename (D:\RSSHub\node_modules\tsx\dist\cjs\index.cjs:1:1729)
    at Module._load (node:internal/modules/cjs/loader:985:27)
    at Module.require (node:internal/modules/cjs/loader:1235:19)
    at require (node:internal/modules/helpers:176:18)
    at path (d:\RSSHub\lib\registry.ts:41:22)
    at Object.<anonymous> (d:\RSSHub\lib\registry.ts:94:8)
    at Module._compile (node:internal/modules/cjs/loader:1376:14)
    at Object.S (D:\RSSHub\node_modules\tsx\dist\cjs\index.cjs:1:1292)
    at Module.load (node:internal/modules/cjs/loader:1207:32)
    at Module._load (node:internal/modules/cjs/loader:1023:12)
    at Module.require (node:internal/modules/cjs/loader:1235:19)
    at require (node:internal/modules/helpers:176:18)
    at registry (d:\RSSHub\lib\app.ts:19:22)
    at Object.<anonymous> (d:\RSSHub\lib\app.ts:44:16)
    at Module._compile (node:internal/modules/cjs/loader:1376:14) {
  code: 'MODULE_NOT_FOUND',
  requireStack: [
    'D:\MyProgram\RSSHub\RSSHub\lib\registry.ts',
    'D:\MyProgram\RSSHub\RSSHub\lib\app.ts',
    'D:\MyProgram\RSSHub\RSSHub\lib\index.ts'
  ]
}

Node.js v20.10.0
error Command failed with exit code 1.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.


almost same here

**Comment by Leegenux** [comment]
Linux上也遇到了同样的问题，当前在`master`的`2c94fc9cb736b5320b4164cd8d8b5beb3205ea4b`这个commit：
```
- /home/leegenux/RSSHub/lib/index.ts
     at Module._resolveFilename (node:internal/modules/cjs/loader:1144:15) 
     ...
     at Module._compile (node:internal/modules/cjs/loader:1376:14)
  code: 'MODULE_NOT_FOUND',
```

**Comment by frankcwl** [comment]
```
npm start

> rsshub@1.0.0 start
> cross-env NODE_ENV=production tsx lib/index.ts

node:internal/modules/cjs/loader:1134
  const err = new Error(message);
              ^

Error: Cannot find module '../assets/build/routes.json'
Require stack:
- C:\RSSHub\lib\registry.ts
- C:\RSSHub\lib\app.ts
- C:\RSSHub\lib\index.ts
    at Module._resolveFilename (node:internal/modules/cjs/loader:1134:15)
    at a._resolveFilename (C:\RSSHub\node_modules\tsx\dist\cjs\index.cjs:1:1729)
    at Module._load (node:internal/modules/cjs/loader:975:27)
    at Module.require (node:internal/modules/cjs/loader:1225:19)
    at require (node:internal/modules/helpers:177:18)
    at path (c:\RSSHub\lib\registry.ts:41:22)
    at Object.<anonymous> (c:\RSSHub\lib\registry.ts:94:8)
    at Module._compile (node:internal/modules/cjs/loader:1356:14)
    at Object.S (C:\RSSHub\node_modules\tsx\dist\cjs\index.cjs:1:1292)
    at Module.load (node:internal/modules/cjs/loader:1197:32)
    at Module._load (node:internal/modules/cjs/loader:1013:12)
    at Module.require (node:internal/modules/cjs/loader:1225:19)
    at require (node:internal/modules/helpers:177:18)
    at registry (c:\RSSHub\lib\app.ts:19:22)
    at Object.<anonymous> (c:\RSSHub\lib\app.ts:44:16)
    at Module._compile (node:internal/modules/cjs/loader:1356:14) {
  code: 'MODULE_NOT_FOUND',
  requireStack: [
    'C:\\RSSHub\\lib\\registry.ts',
    'C:\\RSSHub\\lib\\app.ts',
    'C:\\RSSHub\\lib\\index.ts'
  ]
}

Node.js v18.19.1
```
The same error on Windows, node v18.19.1 and npm v10.2.4 after running `npm install` and `npm start`.

**Comment by Rongronggg9** [comment]
All you need to do is:

```sh
npm run build
npm run start
```

or

```sh
npm run dev
```

**Comment by Leegenux** [comment]
> All you need to do is:
> 
> ```shell
> npm run build
> npm run start
> ```
> 
> or
> 
> ```shell
> npm run dev
> ```

Yes it resolves the problem.

**Comment by frankcwl** [comment]
> All you need to do is:
> 
> ```shell
> npm run build
> npm run start
> ```
> 
> or
> 
> ```shell
> npm run dev
> ```

I still have the same problem. Could you take a look at the error log I posted above? Maybe it's not the same as others.

**Comment by huanfe1** [comment]
报错仍然存在 win10 系统

```
$ pnpm run dev

> rsshub@1.0.0 dev D:\code\RSSHub
> cross-env NODE_ENV=dev tsx watch --no-cache lib/index.ts

node:internal/modules/cjs/loader:1077
  const err = new Error(message);    
              ^

Error: Cannot find module '/D:\code\RSSHub\lib\routes/robots.txt.ts'
Require stack:
- D:\code\RSSHub\node_modules\.pnpm\directory-import@3.2.1\node_modules\directory-import\dist\src\import-modules.js
- D:\code\RSSHub\node_modules\.pnpm\directory-import@3.2.1\node_modules\directory-import\dist\src\index.js
- D:\code\RSSHub\lib\registry.ts
- D:\code\RSSHub\lib\app.ts
- D:\code\RSSHub\lib\index.ts
    at Module._resolveFilename (node:internal/modules/cjs/loader:1077:15)
    at a._resolveFilename (D:\code\RSSHub\node_modules\.pnpm\tsx@4.7.1\node_modules\tsx\dist\cjs\index.cjs:1:1729)
    at Module._load (node:internal/modules/cjs/loader:922:27)
    at Module.require (node:internal/modules/cjs/loader:1143:19)
    at require (node:internal/modules/cjs/helpers:119:18)
    at importModule (D:\code\RSSHub\node_modules\.pnpm\directory-import@3.2.1\node_modules\directory-import\dist\src\import-modules.js:68:28)   
    at Object.syncHandler [as sync] (D:\code\RSSHub\node_modules\.pnpm\directory-import@3.2.1\node_modules\directory-import\dist\src\import-modules.js:21:34)
    at importModules (D:\code\RSSHub\node_modules\.pnpm\directory-import@3.2.1\node_modules\directory-import\dist\src\import-modules.js:84:40)  
    at directoryImport (D:\code\RSSHub\node_modules\.pnpm\directory-import@3.2.1\node_modules\directory-import\dist\src\index.js:17:45)
    at path (d:\code\RSSHub\lib\registry.ts:44:19)
    at Object.<anonymous> (d:\code\RSSHub\lib\registry.ts:121:16)
    at Module._compile (node:internal/modules/cjs/loader:1256:14)
    at Object.S (D:\code\RSSHub\node_modules\.pnpm\tsx@4.7.1\node_modules\tsx\dist\cjs\index.cjs:1:1292)
    at Module.load (node:internal/modules/cjs/loader:1119:32)
    at Module._load (node:internal/modules/cjs/loader:960:12)
    at Module.require (node:internal/modules/cjs/loader:1143:19) {
  code: 'MODULE_NOT_FOUND',
  requireStack: [
    'D:\\code\\RSSHub\\node_modules\\.pnpm\\directory-import@3.2.1\\node_modules\\directory-import\\dist\\src\\import-modules.js',
    'D:\\code\\RSSHub\\node_modules\\.pnpm\\directory-import@3.2.1\\node_modules\\directory-import\\dist\\src\\index.js',
    'D:\\code\\RSSHub\\lib\\registry.ts',
    'D:\\code\\RSSHub\\lib\\app.ts',
    'D:\\code\\RSSHub\\lib\\index.ts'
  ],
  includeSubdirectories: true,
  importMode: 'sync',
  importPattern: /\.ts$/,
  limit: Infinity,
  callerFilePath: '/',
  callerDirectoryPath: '',
  targetDirectoryPath: 'D:\\code\\RSSHub\\lib\\routes',
  callback: undefined
}

Node.js v18.18.0
```

好像是 `directory-import` 包的问题

**Comment by NightSpaceC** [comment]
`directory-import`似乎在路径前添加了'`/`'

我在`node_modules\directory-import\dist\src\import-modules.js:21`改为了

```js
const isModuleImported = importModule(filePath.slice(1), index, options, modules);
```

`build`的时看上去这个问题已被解决，但是产生了另一个错误

```
yarn run v1.22.21
$ tsx scripts/workflow/build-routes.ts
[2024-03-17T23:47:26.890] [INFO] - [Running gramJS version 2.20.1]
node:internal/modules/cjs/loader:1048
  const err = new Error(message);
              ^

Error: Cannot find module 'tosource'
Require stack:
- C:\xxx\RSSHub\scripts\workflow\build-routes.ts
    at Module._resolveFilename (node:internal/modules/cjs/loader:1048:15)
    at a._resolveFilename (C:\xxx\RSSHub\node_modules\tsx\dist\cjs\index.cjs:1:1729)
    at Module._load (node:internal/modules/cjs/loader:901:27)
    at Module.require (node:internal/modules/cjs/loader:1115:19)
    at require (node:internal/modules/helpers:130:18)
    at fs (c:\xxx\RSSHub\scripts\workflow\build-routes.ts:6:22)
    at Object.<anonymous> (c:\xxx\RSSHub\scripts\workflow\build-routes.ts:131:1)
    at Module._compile (node:internal/modules/cjs/loader:1241:14)
    at Object.S (C:\xxx\RSSHub\node_modules\tsx\dist\cjs\index.cjs:1:1292)
    at Module.load (node:internal/modules/cjs/loader:1091:32)
    at Module._load (node:internal/modules/cjs/loader:938:12)
    at cjsLoader (node:internal/modules/esm/translators:284:17)
    at ModuleWrap.<anonymous> (node:internal/modules/esm/translators:234:7)
    at ModuleJob.run (node:internal/modules/esm/module_job:217:25)
    at async ModuleLoader.import (node:internal/modules/esm/loader:316:24)
    at async loadESM (node:internal/process/esm_loader:34:7)
    at async handleMainPromise (node:internal/modules/run_main:66:12) {
  code: 'MODULE_NOT_FOUND',
  requireStack: [
    'C:\\xxx\\RSSHub\\scripts\\workflow\\build-routes.ts'
  ]
}

Node.js v20.9.0
error Command failed with exit code 1.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
```

我用`npm install -g tosource`添加了`tosource`，但没有效果

**Comment by TonyRL** [comment]
Closed in #14992

**Comment by ZionPi** [comment]
I came across this same issue even when I `npm run build`.
After I have checked the script of build as ` "build": "tsx scripts/workflow/build-routes.ts",`.
Then I `npm install tsx`,problem solved.
